utils/usage/usage.go

```diff
@@ -122,6 +134,23 @@ func (ur *UsageReporter) reportToEcosystem(features ...ReportFeature) (err error
 	return ecosysusage.SendEcosystemUsageReports(reports...)
 }
 
+func (ur *UsageReporter) reportToXray(features ...ReportFeature) (err error) {
+	if ur.serverDetails.XrayUrl == "" {
+		err = errorutils.CheckErrorf("Xray Url is not set.")
+		return
+	}
+	serviceManager, err := xrayutils.CreateXrayServiceManager(ur.serverDetails)
+	if err != nil {
+		return
+	}
+	events := ur.convertAttributesToXrayEvents(features...)
+	if len(events) == 0 {
+		err = errorutils.CheckErrorf("Nothing to send.")
+		return
+	}
```

Comment:
Can we run `if len(features) == 0` check before reporting to "artifactory", "xray", and "ecosystem"?

Reply:
each type of report has different attributes and limitations, even if features > 0 there could be an option for some converted format to be empty, added a general check before reporting and moved this check for each report to be the first thing